ipv6/address.go

```diff
@@ -0,0 +1,147 @@
+package ipv6
+
+import (
+	"fmt"
+	"net"
+)
+
+const (
+	// SIZE is the number of bits that an IPv6 address takes
+	SIZE int = 128
+)
+
+// Address represents an IPv6 address
+type Address struct {
+	ui uint128
+}
+
+// AddressFromUint64 returns the IPv6 address from its two 64 bit unsigned representation
+func AddressFromUint64(high uint64, low uint64) Address {
+	return Address{uint128{high, low}}
+}
+
+// AddressFromBytes returns the IPv6 address from bytes
+func AddressFromBytes(s []byte) Address {
+	return Address{Uint128FromBytes(s)}
+}
+
+// AddressFromStdIP converts
+func AddressFromStdIP(ip net.IP) (Address, error) {
+	return fromSlice(ip)
+}
+
+// ParseAddress returns the ip represented by `addr` in dotted-quad notation. If
+// the address cannot be parsed, then error is non-nil and the address returned
+// must be ignored.
+func ParseAddress(address string) (Address, error) {
+	netIP := net.ParseIP(address)
+	if netIP == nil {
+		return Address{}, fmt.Errorf("failed to parse IPv6: %s", address)
+	}
+	return AddressFromStdIP(netIP)
+}
+
+// MinAddress returns the address, a or b, which comes first in lexigraphical order
+func MinAddress(a, b Address) Address {
+	if a.LessThan(b) {
+		return a
+	}
+	return b
+}
+
+// MaxAddress returns the address, a or b, which comes last in lexigraphical order
+func MaxAddress(a, b Address) Address {
+	if a.LessThan(b) {
+		return b
+	}
+	return a
+}
+
+// ToStdIP returns a net.IP representation of the address which always has 4 bytes
+func (me Address) ToStdIP() net.IP {
+	ip := me.toBytes()
+	return ip[:]
+}
+
+// Equal reports whether this IPv6 address is the same as other
+func (me Address) Equal(other Address) bool {
+	return me.ui.Equal(other.ui)
+}
+
+// LessThan reports whether this IPv6 address comes strictly before `other`
+// lexigraphically.
+func (me Address) LessThan(other Address) bool {
```

Comment:
ditto from MinAddress. I decided not to export any ordering methods on the types at this time.